thunder/core/jit_ext.py

```diff
@@ -1576,7 +1576,7 @@ def from_opaque(provenance, *, new_output=False):
             raise NotImplementedError(f"unpacking from OPAQUE {fn.value} {provenance}")
 
         def from_provenance(provenance, *, new_output=False):
-            if hasattr(provenance, "proxy"):
```

Comment:
The `hasattr` here seems fine? If you want to use `getattr`, a pattern like:

```
if (p := getattr(provenance, 'proxy', None)) is not None:
  return p
```

Might make more sense, as the `proxy` attribute is only acquired once

Reply:
I changed it to `getattr` because we may later set `p.history.proxy = None` below on line 1621.

I have just learned the overhead of `getattr` is not negligible. Thanks for your suggestion!

Reply:
@jjsjann123 and I were just talking about the cost of accessing an attribute of an object. It probably takes between .1us and 1us to do so